extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -28,27 +87,371 @@ export class LaterpayVendor {
    * @param {!AccessService} accessService
    */
   constructor(accessService) {
-    /** @private @const */
+    /** @const @private {!AccessService} */
     this.accessService_ = accessService;
-    // TODO: implement
+
+    /** @const @private {!Window} */
+    this.win_ = this.accessService_.win;
+
+    /** @const @private {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportForDoc(this.win_.document);
+
+    /** @const @private {!LaterpayConfigDef} */
+    this.laterpayConfig_ = this.accessService_.getAdapterConfig();
+
+    /** @private {?PurchaseConfigDef} */
+    this.purchaseConfig_ = null;
+
+    /** @private {?Function} */
+    this.purchaseButtonListener_ = null;
+
+    /** @private {?Function} */
+    this.alreadyPurchasedListener_ = null;
+
+    /** @const @private {!Array<Event>} */
+    this.purchaseOptionListeners_ = [];
+
+    /** @private {!boolean} */
+    this.containerEmpty_ = true;
+
+    /** @private {?Node} */
+    this.selectedPurchaseOption_ = null;
+
+    /** @private {?Node} */
+    this.purchaseButton_ = null;
+
+    const configUrl = (
+      (getMode().localDev || getMode().development) &&
+      this.laterpayConfig_.configUrl
+    ) ? this.laterpayConfig_.configUrl
+      : CONFIG_URL;
+
+    /** @private {string} */
+    this.currentLocale_ = this.laterpayConfig_.locale || 'en';
+
+    /** @private {Object} */
+    this.i18n_ = Object.assign(DEFAULT_MESSAGES,
```

Comment:
This will mutate `DEFAULT_MESSAGES`. You can use `Object.assign({}, DEFAULT_MESSAGES, ...)` to avoid.